### ISSUE TITLE
incorrect 'incorrect state' message on agent ingest

### ISSUE THREAD

**Issue by andre-merzky** [radical-cybertools/radical.pilot#1273]
Agent gets unit like this:

```log
2017-03-09 00:05:05,535: agent_0             : 
    MainProcess                     : 
    agent_0.idler._check_units_cb: ERROR   :  
    === invalid state: {u'_id': u'unit.000001',
  u'state':   u'UMGR_STAGING_INPUT',
  u'states': [u'NEW',
              u'UMGR_SCHEDULING_PENDING',
              u'UMGR_SCHEDULING',
              u'UMGR_STAGING_INPUT_PENDING',
              u'UMGR_STAGING_INPUT',
              u'AGENT_STAGING_INPUT_PENDING',
              u'UMGR_STAGING_INPUT_PENDING', 
              u'UMGR_STAGING_INPUT'],
```

so we need to collapse the state from `states` before checking the `state` field.

**Comment by andre-merzky** [comment]
see #1271

**Comment by andre-merzky** [comment]
This should be solved with merging #1271
